### PR TITLE
Convert to TypeScript

### PR DESCRIPTION
**Pull request by mike-north**
Fixes #3

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "typescript.tsdk": "node_modules/typescript/lib"
```
Comment (josemarluedke) [inline .vscode/settings.json]: Should this change be here?

File: package.json [hunk 2]
```diff
@@ -17,9 +17,11 @@
     "lint:js": "eslint .",
     "start": "ember serve",
     "test": "ember test",
-    "test:all": "ember try:each"
+    "test:all": "ember try:each",
+    "prepublish": "ember ts:precompile"
```
Comment (josemarluedke) [inline package.json]: The default `ember-cli-typescript` blueprint adds the follow tasks to deal with publishing: 

```
    "prepublishOnly": "ember ts:precompile",
    "postpublish": "ember ts:clean"
```

File: tracked.d.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export declare function tracked(target: object, propertyKey: string | symbol, descriptor?: PropertyDescriptor): any;
```
Comment (josemarluedke) [inline tracked.d.ts]: Should this file be checked into git?

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+declare class SparklesComponent<T = object> {
```
Comment (josemarluedke) [inline index.d.ts]: Should this file be checked into git?

File: component-managers/sparkles.d.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
+/// <reference types="ember" />
```
Comment (josemarluedke) [inline component-managers/sparkles.d.ts]: Same as the others. Should this file be checked into git?

**Comment by rwjblue** [comment]
Awesome! Let’s give it a whirl...

**Pull request merged**